cime_config/buildnml

```diff
@@ -1,72 +1,68 @@
 #!/usr/bin/env python
-"""Namelist creator for CIME's driver.
+
+"""Namelist creator for NEMS's driver.
 """
-import os, sys
 
-_CIMEROOT = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..","..","..","..")
-sys.path.append(os.path.join(_CIMEROOT, "scripts", "Tools"))
+# Typically ignore this.
+# pylint: disable=invalid-name
+
+# Disable these because this is our standard setup
+# pylint: disable=wildcard-import,unused-wildcard-import,wrong-import-position
+
+import os, sys, copy, shutil
+
+CIMEROOT = os.environ.get("CIMEROOT")
+if CIMEROOT is None:
+    raise SystemExit("ERROR: must set CIMEROOT environment variable")
+sys.path.append(os.path.join(CIMEROOT, "scripts", "Tools"))
 
-import shutil, glob, itertools
 from standard_script_setup import *
 from CIME.case import Case
 from CIME.nmlgen import NamelistGenerator
-from CIME.utils import expect
-from CIME.utils import get_model, get_time_in_seconds, get_timestamp
+from CIME.namelist import parse
 from CIME.buildnml import create_namelist_infile, parse_input
-from CIME.XML.files import Files
-#pylint: disable=undefined-variable
+from CIME.utils import expect, safe_copy, symlink_force, ls_sorted_by_mtime
+
 logger = logging.getLogger(__name__)
 
+# returns active components
+def active_components(case):
```

Comment:
comps = [case.get_value("COMP_{}".format(comp_class)) for comp_class in case.get_values("COMP_CLASSES")]

Reply:
my function returns only `['ufsatm%v0nocp']` but yours `['cpl', 'ufsatm']`. So, I keep it as it is.